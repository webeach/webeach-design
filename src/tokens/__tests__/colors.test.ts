import { DarkModeColor, LightModeColor } from '../colors';

import { colorMapSchema } from './schemas/colorMapSchema';

describe('Colors validation', () => {
  test('Validate DarkModeColor', () => {
    expect(() => colorMapSchema.parse(DarkModeColor)).not.toThrow();
  });

  test('Validate LightModeColor', () => {
    expect(() => colorMapSchema.parse(LightModeColor)).not.toThrow();
  });

  test('Snapshot DarkModeColor', () => {
    expect(DarkModeColor).toMatchSnapshot();
  });

  test('Snapshot LightModeColor', () => {
    expect(LightModeColor).toMatchSnapshot();
  });
});
