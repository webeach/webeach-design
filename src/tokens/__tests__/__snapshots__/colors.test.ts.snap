// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors validation Snapshot DarkModeColor 1`] = `
{
  "danger1": "hsl(0 100% 10%)",
  "danger10": "hsl(0 100% 85%)",
  "danger2": "hsl(0 100% 15%)",
  "danger3": "hsl(0 100% 20%)",
  "danger4": "hsl(0 100% 25%)",
  "danger5": "hsl(0 100% 30%)",
  "danger6": "hsl(0 67% 48%)",
  "danger7": "hsl(0 79% 54%)",
  "danger8": "hsl(0 100% 60%)",
  "danger9": "hsl(0 100% 70%)",
  "neutral1": "hsl(0 0% 0%)",
  "neutral10": "hsl(0 0% 100%)",
  "neutral2": "hsl(0 0% 7%)",
  "neutral3": "hsl(240 3% 11%)",
  "neutral4": "hsl(240 2% 18%)",
  "neutral5": "hsl(240 2% 23%)",
  "neutral6": "hsl(240 1% 29%)",
  "neutral7": "hsl(240 1% 39%)",
  "neutral8": "hsl(240 2% 57%)",
  "neutral9": "hsl(240 1% 83%)",
  "primary1": "hsl(300 45% 11%)",
  "primary10": "hsl(299 77% 90%)",
  "primary2": "hsl(300 49% 15%)",
  "primary3": "hsl(300 48% 20%)",
  "primary4": "hsl(300 49% 28%)",
  "primary5": "hsl(300 47% 35%)",
  "primary6": "hsl(300 43% 49%)",
  "primary7": "hsl(302 59% 63%)",
  "primary8": "hsl(302 67% 72%)",
  "primary9": "hsl(301 72% 80%)",
  "secondary1": "hsl(200 71% 13%)",
  "secondary10": "hsl(196 86% 88%)",
  "secondary2": "hsl(198 70% 18%)",
  "secondary3": "hsl(197 69% 25%)",
  "secondary4": "hsl(201 70% 34%)",
  "secondary5": "hsl(201 70% 42%)",
  "secondary6": "hsl(202 71% 47%)",
  "secondary7": "hsl(200 85% 63%)",
  "secondary8": "hsl(199 86% 70%)",
  "secondary9": "hsl(198 85% 79%)",
  "success1": "hsl(120 36% 10%)",
  "success10": "hsl(120 52% 88%)",
  "success2": "hsl(120 32% 15%)",
  "success3": "hsl(120 33% 21%)",
  "success4": "hsl(120 33% 26%)",
  "success5": "hsl(120 40% 35%)",
  "success6": "hsl(123 41% 45%)",
  "success7": "hsl(120 49% 62%)",
  "success8": "hsl(120 50% 71%)",
  "success9": "hsl(120 52% 80%)",
  "warning1": "hsl(40 100% 10%)",
  "warning10": "hsl(37 92% 80%)",
  "warning2": "hsl(38 100% 15%)",
  "warning3": "hsl(40 100% 20%)",
  "warning4": "hsl(41 100% 25%)",
  "warning5": "hsl(42 100% 30%)",
  "warning6": "hsl(36 75% 49%)",
  "warning7": "hsl(35 89% 55%)",
  "warning8": "hsl(34 89% 63%)",
  "warning9": "hsl(36 90% 72%)",
}
`;

exports[`Colors validation Snapshot LightModeColor 1`] = `
{
  "danger1": "hsl(0 100% 97%)",
  "danger10": "hsl(0 100% 25%)",
  "danger2": "hsl(0 100% 92%)",
  "danger3": "hsl(0 100% 85%)",
  "danger4": "hsl(0 100% 75%)",
  "danger5": "hsl(0 100% 65%)",
  "danger6": "hsl(0 100% 60%)",
  "danger7": "hsl(0 79% 54%)",
  "danger8": "hsl(0 67% 48%)",
  "danger9": "hsl(0 67% 42%)",
  "neutral1": "hsl(0 0% 100%)",
  "neutral10": "hsl(0 0% 0%)",
  "neutral2": "hsl(0 0% 98%)",
  "neutral3": "hsl(0 0% 96%)",
  "neutral4": "hsl(0 0% 88%)",
  "neutral5": "hsl(0 0% 80%)",
  "neutral6": "hsl(0 0% 66%)",
  "neutral7": "hsl(0 0% 53%)",
  "neutral8": "hsl(0 0% 40%)",
  "neutral9": "hsl(0 0% 27%)",
  "primary1": "hsl(300 83% 95%)",
  "primary10": "hsl(294 43% 22%)",
  "primary2": "hsl(299 77% 90%)",
  "primary3": "hsl(301 72% 80%)",
  "primary4": "hsl(302 71% 71%)",
  "primary5": "hsl(302 72% 62%)",
  "primary6": "hsl(301 53% 55%)",
  "primary7": "hsl(301 42% 49%)",
  "primary8": "hsl(299 41% 40%)",
  "primary9": "hsl(297 41% 31%)",
  "secondary1": "hsl(203 85% 95%)",
  "secondary10": "hsl(208 79% 29%)",
  "secondary2": "hsl(203 85% 89%)",
  "secondary3": "hsl(204 85% 84%)",
  "secondary4": "hsl(205 85% 79%)",
  "secondary5": "hsl(204 84% 73%)",
  "secondary6": "hsl(201 84% 55%)",
  "secondary7": "hsl(202 71% 49%)",
  "secondary8": "hsl(203 72% 42%)",
  "secondary9": "hsl(205 75% 36%)",
  "success1": "hsl(120 52% 94%)",
  "success10": "hsl(123 46% 34%)",
  "success2": "hsl(120 52% 88%)",
  "success3": "hsl(120 52% 80%)",
  "success4": "hsl(120 50% 71%)",
  "success5": "hsl(120 49% 62%)",
  "success6": "hsl(122 39% 49%)",
  "success7": "hsl(123 41% 45%)",
  "success8": "hsl(123 43% 39%)",
  "success9": "hsl(122 39% 49%)",
  "warning1": "hsl(46 100% 94%)",
  "warning10": "hsl(32 72% 27%)",
  "warning2": "hsl(40 93% 89%)",
  "warning3": "hsl(37 92% 80%)",
  "warning4": "hsl(36 90% 72%)",
  "warning5": "hsl(34 89% 63%)",
  "warning6": "hsl(35 89% 55%)",
  "warning7": "hsl(36 75% 49%)",
  "warning8": "hsl(37 75% 44%)",
  "warning9": "hsl(34 74% 35%)",
}
`;
